CMakeLists.txt

```diff
@@ -1,23 +1,65 @@
 cmake_minimum_required(VERSION 3.12) #Python3 package locator only added in 3.12
 
+project(cmakedoc DESCRIPTION "CMake automatic documentation generator" LANGUAGES NONE) #Not using any compiled languages so don't list, removes check for compilers
+
 find_package(Python3 COMPONENTS Interpreter)
 
+#[[[
+# Custom function to locate a Python module. Requires the Python interpreter to be located first.
+#
+# Sets PY_${module_upper} to the module file's location if found.
+#
+# :param module: Python module name to locate
+# :param REQUIRED: Optional paramater, specifies that this module is required for continued execution.
+# Sets ${module}_FIND_REQUIRED to true.
+#
+#]]
+function(find_python_module module)
+	string(TOUPPER ${module} module_upper)
+	if(NOT PY_${module_upper})
+		if(ARGC GREATER 1 AND ARGV1 STREQUAL "REQUIRED")
```

Comment:
A better way to handle optional function arguments in CMake is with [cmake_parse_arguments](https://cmake.org/cmake/help/latest/command/cmake_parse_arguments.html?highlight=parse_argumen#command:cmake_parse_arguments). That'll help you out later if you need to add additional options later.

Reply:
Should be fixed now